scripts/powershell/Verify-And-Merge-PRs.ps1

```diff
@@ -0,0 +1,145 @@
+param(
+    [Parameter(Mandatory = $true)]
+    $AuthToken,
+
+    [Parameter(Mandatory = $true)]
+    $PRDataArtifactPath
+)
+
+$PRObjs = [System.Collections.ArrayList]::new()
+$MergedPRs = [System.Collections.ArrayList]::new()
+$ClosedPRs = [System.Collections.ArrayList]::new()
+$UnexpectedState = [System.Collections.ArrayList]::new()
+$OpenAndCleanPRs = [System.Collections.ArrayList]::new()
+$OpenButDirtyPRs = [System.Collections.ArrayList]::new()
+
+
+$GitHubAPIUrlBase = "https://api.github.com/repos/"
+$Headers = @{
+    Authorization = "bearer $AuthToken"
+}
+
+function Merge-PRs()
+{
+    if (($OpenButDirtyPRs.Count -gt 0) -or ($ClosedPRs.Count -gt 0) -or ($UnexpectedState.Count -gt 0))
+    {
+        Write-Output "The following PRs are either closed, or in an Un-mergeable state"
```

Comment:
For this type of logging I suggest Write-Host. While Write-Output works in it sometimes doesn't do what you might expect, For example if you pipe the output of this function to something else it will essentially write it to the pipeline instead of to the console.